src/api/artifact/controller.go

```diff
@@ -453,6 +459,10 @@ func (c *controller) Walk(ctx context.Context, root *Artifact, walkFn func(*Arti
 
 		artifact := elem.Value.(*Artifact)
 		if err := walkFn(artifact); err != nil {
+			if err == ErrBreak {
```

Comment:
How about defining the walkFn as `func(*Artifact)(continue bool, err error)`, the first returning parameter indicates whether the walk should continue or not

Reply:
`func(*Artifact) err` is just ok for this, the `filepath.Walk` returns `filepath.SkipDir` to skip the remaining files in the containing directory.